pokemon.go

```diff
@@ -101,10 +101,20 @@ func (p Pokemon) String() string {
 		p.Gender, p.Level, p.CurrentHP, p.Stats[STAT_HP])
 }
 
+// Applies damage to a Pokemon. May cause Pokemon to faint.
+func (p *Pokemon) Damage(amount uint) {
+	if p.CurrentHP >= amount {
+		p.CurrentHP -= amount
+	} else {
+		// prevent underflow
+		p.CurrentHP = 0
+	}
+}
+
 // Restore HP to a Pokemon. Can also be used to revive a fainted Pokemon.
-func (p *Pokemon) RestoreHP(amount uint) Transaction {
+func (p *Pokemon) Heal(amount uint) {
```

Comment:
oh its interesting that you can do both. with our current set up, would we be able to use a potion on a fainted Pokemon?

Reply:
probably

Reply:
is that a fix that is relevant to this PR or should it go into a new issue?

Reply:
thought about it a bit, I think it can be resolved in its own issue